### PR TITLE
GST-101 put timed text usage in single lib.

### PR DESCRIPTION
**Pull request by noneimb**
Move parsing from `gstttmlparse `to `gstttmlcommon `to avoid multiple instances of static variables from `timedText` which causes ambiguity and usage of wrong static variable value.

### CODE REVIEW

File: gst/parser/gstttmlparse.c [hunk 1]
```diff
@@ -1607,8 +1607,9 @@ handle_buffer (GstTtmlParse * self, GstBuffer * buf)
   if (g_strcmp0 (self->subtitle_codec, "EBUTT") == 0) {
     GList *subtitle;
     GTimer *timer = g_timer_new ();
-    CParser *ttml_parser = parse_ttml (self->textbuf->str, self->forced_only);
-    if (ttml_parser == NULL) {
+
+    CParser* subs_parser = create_subs_parser();
```
Comment (khwaaj) [inline gst/parser/gstttmlparse.c:1611]: I guess we should check for `NULL` here, no?

**Review by khwaaj — APPROVED** [review]
Should probably add a `NULL` check after parser creation, otherwise 👍

**Comment by khwaaj** [comment]
Also please check the two minor fixes I added before merging.

File: gst/parser/gstttmlparse.c (hunk 1 above)
Comment (noneimb) [reply gst/parser/gstttmlparse.c:1611]: Added that

**Pull request merged**